### PR TITLE
IPBlock convenience functions

### PR DESCRIPTION
**Pull request by YairSlobodin1**

### CODE REVIEW

File: pkg/netset/ipblock.go [hunk 1]
```diff
@@ -396,6 +464,18 @@ func GetCidrAll() *IPBlock {
 	return res
 }
 
+// GetFirstIPAdress returns IPBlock object of 0.0.0.0
+func GetFirstIPAdress() *IPBlock {
```
Comment (zivnevo) [inline pkg/netset/ipblock.go:468]:
```suggestion
func GetFirstIPAddress() *IPBlock {
```
No spell checker in your IDE?

File: pkg/netset/ipblock.go [hunk 2]
```diff
@@ -396,6 +464,18 @@ func GetCidrAll() *IPBlock {
...
+	res, _ := IPBlockFromIPAddress(FirstIPAddressString)
+	return res
+}
+
+// GetLastIPAdress returns IPBlock object of 255.255.255.255
+func GetLastIPAdress() *IPBlock {
```
Comment (zivnevo) [inline pkg/netset/ipblock.go:474]:
```suggestion
func GetLastIPAddress() *IPBlock {
```

File: pkg/netset/ipblock.go [hunk 3]
```diff
@@ -156,6 +164,18 @@ func (b *IPBlock) Split() []*IPBlock {
 	return res
 }
 
+// SplitToCidrs returns a set of Cidrs
```
Comment (zivnevo) [inline pkg/netset/ipblock.go:167]:
```suggestion
// SplitToCidrs returns a slice of IPBlocks, each representing a single CIDR
```

File: pkg/netset/ipblock.go [hunk 4]
```diff
@@ -156,6 +164,18 @@ func (b *IPBlock) Split() []*IPBlock {
...
+func (b *IPBlock) SplitToCidrs() []*IPBlock {
+	cidrs := make([]*IPBlock, 0)
+	for _, ipRange := range b.ipRange.Intervals() {
+		for _, cidrString := range intervalToCidrList(ipRange) {
+			ipblock, _ := IPBlockFromCidrOrAddress(cidrString)
```
Comment (zivnevo) [inline pkg/netset/ipblock.go:172]:
```suggestion
			ipblock, _ := IPBlockFromCidr(cidrString)
```
Save the check for the CIDR separator

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/netset/ipblock.go [hunk 5]
```diff
@@ -415,3 +495,11 @@ func (b *IPBlock) String() string {
 	}
 	return b.ToCidrListString()
 }
+
+// TouchingIPRanges returns true if this and other ipblocks objects are touching
+func (b *IPBlock) TouchingIPRanges(other *IPBlock) bool {
+	u := b.Copy().Union(b)
```
[the remark this answers is not in the document]
Comment (YairSlobodin1) [reply pkg/netset/ipblock.go:501]: fixed

**Review by zivnevo — CHANGES_REQUESTED** [review]
Docstring for renamed functions has to be updated too

File: pkg/netset/ipblock.go [hunk 6]
```diff
@@ -415,3 +495,12 @@ func (b *IPBlock) String() string {
...
 	return b.ToCidrListString()
 }
+
+// TouchingIPRanges returns true if this and other ipblocks objects are touching.
+// assumption: both IPBlocks are a single IP range
+func (b *IPBlock) TouchingIPRanges(other *IPBlock) (bool, error) {
+	if len(b.Split()) != 1 || len(other.Split()) != 1 {
+		return false, fmt.Errorf("both ipblocks should be a single IP range")
+	}
+	return (!b.Overlap(other) && len(b.Union(other).Split()) == 1), nil
```
Comment (zivnevo) [inline pkg/netset/ipblock.go:505]:
```suggestion
	if b.ipRange.NumIntervals() != 1 || other.ipRange.NumIntervals() != 1 {
		return false, fmt.Errorf("both ipblocks should be a single IP range")
	}
	return (!b.Overlap(other) && b.Union(other).ipRange.NumIntervals() == 1), nil
```

**Review by zivnevo — CHANGES_REQUESTED** [review]
Please add tests for exported functions in `ipblock_test.go`

**Review by zivnevo — APPROVED** [review]

**Pull request merged**